pkg/operator/vspherecontroller/operator_removal.go

```diff
@@ -0,0 +1,135 @@
+package vspherecontroller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"github.com/openshift/vmware-vsphere-csi-driver-operator/assets"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+)
+
+// prefix for conditions used by all conditions
+const (
+	vsphereOperatorPrefix = "VMwareVSphere"
+)
+
+var untouchableConditions = sets.New[string](
+	"VMwareVSphereControllerDegraded",
+	"VMwareVSphereDriverFeatureConfigControllerDegraded",
+	"VMWareVSphereDriverServiceMonitorControllerDegraded",
+)
+
+func (c *VSphereController) removeOperands(ctx context.Context, status *operatorapi.OperatorStatus) error {
+	assetsToBeDeleted := []string{
+		"rbac/privileged_role.yaml",
+		"rbac/controller_privileged_binding.yaml",
+		"rbac/node_privileged_binding.yaml",
+		"rbac/csi_driver_controller_role.yaml",
+		"rbac/csi_driver_controller_binding.yaml",
+		"rbac/csi_driver_node_cluster_role.yaml",
+		"rbac/csi_driver_node_cluster_binding.yaml",
+		"rbac/csi_driver_node_role.yaml",
+		"rbac/csi_driver_node_binding.yaml",
+		"rbac/main_attacher_binding.yaml",
+		"rbac/main_provisioner_binding.yaml",
+		"rbac/volumesnapshot_reader_provisioner_binding.yaml",
+		"rbac/main_resizer_binding.yaml",
+		"rbac/storageclass_reader_resizer_binding.yaml",
+		"rbac/kube_rbac_proxy_role.yaml",
+		"rbac/kube_rbac_proxy_binding.yaml",
+		"rbac/prometheus_role.yaml",
+		"rbac/prometheus_rolebinding.yaml",
+		"rbac/main_snapshotter_binding.yaml",
+		"rbac/lease_leader_election_role.yaml",
+		"rbac/lease_leader_election_rolebinding.yaml",
+		"controller_sa.yaml",
+		"controller_pdb.yaml",
+		"node_sa.yaml",
+		"csidriver.yaml",
+		"service.yaml",
+		"ca_configmap.yaml",
+		// Static assets used by the webhook
+		"webhook/config.yaml",
+		"webhook/sa.yaml",
+		"webhook/service.yaml",
+		"webhook/configuration.yaml",
+		"webhook/rbac/role.yaml",
+		"webhook/rbac/rolebinding.yaml",
+		"webhook/pdb.yaml",
+		// dynamic assets used by controller
+		"volumesnapshotclass.yaml",
```

Comment:
volumesnapshotclass.yaml is static asset, isn't it?

Reply:
So, although it is a static asset, it only gets installed if certain CRD is present.  So I have kept it under dynamic assets.